src/main/cljc/jarl/builtins/objects.cljc

```diff
@@ -14,6 +16,44 @@
     (builtin-object-remove {:args [object (vec (keys ks))]})
     (apply dissoc object ks)))
 
+(defn- subvec? [super sub]
+  (->> (partition (count sub) 1 super)
+       (filter #(= sub %))
+       first
+       seq
+       some?))
+
+(defn- ->key-paths
+  ([x] (into {} (->key-paths x [])))
+  ([x path]
+   (if (map? x)
+     (mapcat #(->key-paths (val %) (conj path (key %))) x)
+     {path x})))
+
+(defn- sub-match? [sub-val sup-val]
+  (cond
+    (and (set? sub-val) (set? sup-val))       (set/subset? sub-val sup-val)
+    (and (vector? sub-val) (vector? sup-val)) (subvec? sup-val sub-val)
+    :else (= sub-val sup-val)))
+
+(defn- submap? [super sub]
+  (let [kp-sub (->key-paths sub)
+        kp-sup (->key-paths super)]
```

Comment:
Don't be afraid of long variable names. I might need glasses, but it took me a second to spot the difference between `kp-sub` and `kp-sup` 😄 .

Reply:
Haha, yeah you're probably right. Using the longer "super" breaks the symmetry though 😅